pkg/vm/vm.go

```diff
@@ -472,7 +500,9 @@ func (v *VM) Step() error {
 
 // step executes one instruction in the given context.
 func (v *VM) step(ctx *Context) error {
+	instruction_offset := v.Context().nextip
 	op, param, err := ctx.Next()
+	v.hooks.onExec(v.GetCurrentScriptHash(), instruction_offset, op)
```

Comment:
`v.GetCurrentScriptHash()` returns script hash *after* instruction execution which is critical for those opcodes that perform context loading/unloading (e.g. `RET` and etc.). Are you sure we need to use not the script hash *before* instruction execution?

Reply:
I'm not against of `GetCurrentScriptHash()` usage. But I think that it should be called before the `op, param, err := ctx.Next()` and saved into a variable. @fyfyrchik, what do you think?

Reply:
Thanks for spotting such a subtle mistake. I've totally missed that

Reply:
Done